honeybee_3dm/config.py

```diff
@@ -0,0 +1,432 @@
+    layer_check = [True for layer in config['layers'] if layer in layers]
+    if len(config['layers']) != layer_check.count(True):
+        raise KeyError(
+            'Only layer names from the Rhino file are allowed in the config file.'
+        )
+    else:
+        return True
+
+
+def check_rad(config):
+    """Checks if radiance modifiers can be applied to objects on rhino layers.
+
+    Args:
+        config: Config dictionary.
+
+    Raises:
+        OSError: If the path to the radiance material file is not a valid path.
+        KeyError: If the config file does not have a key named "radiance_material" in
+            sources and a radiance material is requested in one of the layers in the
+            config file.
+        ValueError: If any of the radiance identifiers mentioned in the config file 
+            are not found in the radiance materials file or do not match the identifiers
+            in the radiance material file.
+
+    Returns:
+        Boolean value of True if no error is raised.
+    """
+
+    # Check-04: Check if radiance modifiers can be imported
+    for layer in config['layers']:
+        if 'radiance_material' in config['layers'][layer]:
+            break
+        if 'sources' in config and config['sources']['radiance_material']:
+            os.path.isfile(config['sources']['radiance_material'])
+            try:
+                modifiers_dict = mat_to_dict(config['sources']['radiance_material'])
+            except OSError:
+                path = config['sources']['radiance_material']
+                raise OSError(
+                    f'The path {path} is not a valid path. Please try' 
+                    ' using double backslashes in the  file path.'
+                        )
+        else:
+            raise KeyError(
+                'Please make sure the config file has the key "radiance_material" in'
+                ' sources.'
+        )
+    
+    # Check-05: Check if all the radiance materials are found in the .mat file
+    rad_mat = [config['layers'][layer][key] for layer in config['layers'] for key \
+        in config['layers'][layer] if key == 'radiance_material']
+    
+    rad_mat_check = [True for mat in rad_mat if mat in modifiers_dict]
+    if len(rad_mat) != rad_mat_check.count(True):
+        raise ValueError(
+            'Please make sure all the radiance materials used in the config file are'
+            ' also found in the radiance material file and names of radiance'
+            ' materials in the config file match the radiance identifiers in the'
+            ' radiance material file.'
+        )
+    else:
+        pass
+
+    return True
+
+
+def check_config(file_3dm, config_path):
+    """Validates the config file and returns it in the form of a dictionary.
+
+    Args:
+        file_3dm: A rhino3dm file object.
+        config_path: A text string for the path to the config file.
+
+    Returns:
+        Config dictionary or None if any of the checks fails.
```

Comment:
Why don't you just work with the `Config` object instead of exporting it as a dictionary?